### PR TITLE
docs: danish language support in configmode

### PR DESCRIPTION
**Pull request by svenfl**
This pull request replaces #390 
I took over and continue the former project of DO9XE

### CODE REVIEW

**Comment by rubo77** [comment]
It doesn't suffice to add a language file `da.po` in the `site/` folder. You need to add danish language support to OpenWRT as well.

so far I tried this in my gluon build folder  after `make update`:

    cp -a packages/luci/modules/luci-base/po/en packages/luci/modules/luci-base/po/da
    for i in package/*/i18n/de.po; do cp -a "$i" "$(echo $i|sed s/de.po/da.po/g)"; done

this should add at least all needed `da.po` files in all packages, so you can start translating everything.

Have a look at the pr for the frensh translation: https://github.com/freifunk-gluon/gluon/pull/448/files

File: docs/site-example/i18n/da.po [hunk 1]
```diff
@@ -0,0 +1,42 @@
+msgid ""
+msgstr ""
+"Content-Type: text/plain; charset=UTF-8\n"
+"Project-Id-Version: PACKAGE VERSION\n"
+"PO-Revision-Date: 2015-03-19 20:28+0100\n"
+"Last-Translator: Sven Thomsen sven@lugfl.de\n"
+"Language-Team: Danish\n"
+"Language: en\n"
```
Comment (rubo77) [inline docs/site-example/i18n/da.po]: Ths should contain `"Language: da\n"`

**Comment by rubo77** [comment]
I think, just copying is not enough, You also have to change that string in all po files, I try this now:

```
cp -a packages/luci/modules/luci-base/po/en packages/luci/modules/luci-base/po/da
sed -i 's/Language: en/Language: da/g' packages/luci/modules/luci-base/po/da/base.po
for i in */*/i18n/de.po */*/*/i18n/de.po; do
  cp -a "$i" "$(echo $i|sed s/de.po/da.po/g)";
  sed -i 's/Language: de/Language: da/g'  "$i"
  sed -i 's/Language-Team: German/Language-Team: Danish/g'  "$i"
done
```

And add this line manually in `gluon/packages/luci/luci.mk`:

    LUCI_LANG.da=Dansk (Danish)

**Comment by NeoRaider** [comment]
@rubo77 The correct command to generate the new PO file is:
```
cd packages/luci/modules/luci-base/po
mkdir da
msginit -o da/base.po -i templates/base.pot -l da
```

As long as there's no Danish locale for luci-base in https://github.com/openwrt/luci/ , we don't even need to think about adding it to Gluon...

**Comment by rubo77** [comment]
Ok, I created an issue: https://github.com/openwrt/luci/issues/785

**Comment by rubo77** [comment]
Or, since Norwegian and Danish is very similar and we probably don't need the norwegian support, we could just use the norwegian language integration as danish.

Norwegian is supported in OpenWRT

will danish be supported natively in LEDE?

should we add the support there instead of OpenWRT?

**Comment by NeoRaider** [comment]
@rubo77, did you even read what I wrote?

1. Don't use cp to create PO files, use msginit!
2. Danish language support is worthless unless it is done properly. At least the base.po should be translated completely, and someone would need to maintain it.

**Comment by NeoRaider** [comment]
@rubo77, neither OpenWrt nor LEDE have anything to do with these translations. They are part of LuCI, which is a mostly separate project used by both OpenWrt and LEDE.

**Comment by rubo77** [comment]
So LEDE also uses this repo? https://github.com/openwrt/luci/

**Comment by rubo77** [comment]
You also have to copy those:

    NEW=da
    BASE=no
    for i in packages/luci/applications/luci-app-*/po/$BASE; do cp -a $i "$(echo $i|sed 's/'$BASE'$/'$NEW'/g')"; done

**Comment by NeoRaider** [comment]
@rubo77 As I wrote before, never use the `cp` command when working with translations.  Starting with a copy is not the way to go. Stop making these comments, they are wrong and confusing. Use `msginit`, `msgmerge` and the tools `build/i18n-*` from the LuCI repo, they will do the work for you.

The previous comment is completely wrong. Tust ignore the applications for now, the base.po is enough. The build should work just fine with missing application translations.

**Comment by rubo77** [comment]
As stated [here](https://github.com/openwrt/luci/pull/790#issuecomment-248552736):

>Gluon seems to need the  `for-15.05` branch instead of master. And new languages will not be introduced here for the old `for-15.05` branch. So you might be better off by crafting a patch in Gluon.

So what would be the best way to introduce danish into gluon?

- [ ] A patch that adds the needed da.po files and adds support for danish in general?
- [ ] A patch that uses an own fork of the  `for-15.05` branch of luci and uses this to build gluon instead?
- [ ] Something completely different

**Comment by NeoRaider** [comment]
Don't care about 15.05, we will switch to the master very soon.

**Comment by NeoRaider** [comment]
As we don't use LuCI anymore, adding a Danish translation is now much easier. If you rebase this and add translation for the new (much smaller) base in gluon-web, we can get this merged.

**Comment by rubo77** [comment]
Now the new msgids gluon-config-mode:novpn",
"gluon-config-mode:altitude-label" and 
 "gluon-config-mode:altitude-help
have to be added

**Comment by rotanid** [comment]
@svenfl do you still plan to finish this PR soon?

**Comment by svenfl** [comment]
I think I need more time. That is difficult at the moment

**Comment by rotanid** [comment]
is there news about the progress?
if there hasn't been any progress until February, we will close this pull request. you're welcome to reopen it afterwards if you plan to continue working on it.

**Comment by svenfl** [comment]
Sorry I had not enough time to continue danish translation
Any help is welcome

**Pull request closed without merging**